src/VueProgram.ts

```diff
@@ -0,0 +1,125 @@
+import fs = require('fs');
+import path = require('path');
+import ts = require('typescript');
+import FilesRegister = require('./FilesRegister');
+import FilesWatcher = require('./FilesWatcher');
+import vueParser = require('vue-parser');
+
+class VueProgram {
+  static loadProgramConfig(configFile: string) {
+    // Add .vue to list of additional extensions to read files from.
+    const extraExtensions = ['vue'];
+    
+    const parseConfigHost: ts.ParseConfigHost = {
+        fileExists: ts.sys.fileExists,
+        readFile: ts.sys.readFile,
+        useCaseSensitiveFileNames: ts.sys.useCaseSensitiveFileNames,
+        readDirectory: (rootDir, extensions, excludes, includes, depth) => {
+            return ts.sys.readDirectory(rootDir, extensions.concat(extraExtensions), excludes, includes, depth);
+        }
+    };
+
+    const parsed = ts.parseJsonConfigFileContent(
+      // Regardless of the setting in the tsconfig.json we want isolatedModules to be false
+      Object.assign(ts.readConfigFile(configFile, ts.sys.readFile).config, { isolatedModules: false }),
+      parseConfigHost,
+      path.dirname(configFile)
+    );
+
+    parsed.options.allowNonTsExtensions = true;
+
+    return parsed;
+  }
+
+  /**
+  * Fixes a non-ts module's name so the file and source text can be properly found later.
+  * Essentially just resolves the module's name from a relative or wildcard path into an absolute path.
+  * Assumes wildcard "@" is [location of tsconfig]/src.
+  */
+  static resolveNonTsModuleName(moduleName: string, containingFile: string, basedir: string): string {
+    if (moduleName.indexOf('@/') === 0) {
+      moduleName = basedir + '/src' + moduleName.substr(1);
+    }
+    else if (moduleName.indexOf('./') === 0 || moduleName.indexOf('../') === 0) {
```

Comment:
please follow project code style and write `else if` next to the closing curly brace :)